lang/cs/Org.Apache.REEF.IMRU.Examples/PipelinedBroadcastReduce/FaultTolerantPipelinedBroadcastAndReduce.cs

```diff
@@ -85,35 +111,54 @@ internal static bool IsEvaluatorFailure(int failureType)
             }
         }
 
+        [NamedParameter(Documentation = "Total number of failures in recovery.", DefaultValue = "2")]
+        internal class TotalNumberOfAllowedFailures : Name<int>
+        {
+        }
+
         /// <summary>
         /// The function is to simulate Evaluator/Task failure for mapper evaluator
         /// </summary>
-        public sealed class TestSenderMapFunction : IMapFunction<int[], int[]>
+        internal sealed class TestSenderMapFunction : IMapFunction<int[], int[]>
         {
             private int _iterations;
             private readonly string _taskId;
             private readonly ISet<string> _taskIdsToFail;
-            private int _failureType;
+            private readonly int _failureType;
             private readonly int _maxRetryInRecovery;
+            private readonly int _totalNumberOfAllowedFailurs;
+            private readonly int _numberOfRetry = 0;
```

Comment:
Better name `_retryIndex` (we're using numberOfSomething to denote count of something, not its serial number)